### ISSUE TITLE
tar_stan_compile fails with default cloud repository

### ISSUE THREAD

**Issue by stuvet** [ropensci/stantargets#54]
## Prework

* [x] Read and agree to the [code of conduct](https://ropensci.org/code-of-conduct/) and [contributing guidelines](https://github.com/ropensci/stantargets/blob/main/CONTRIBUTING.md).
* [x] Confirm that your issue is most likely a genuine bug in `stantargets` and not a known limitation, usage error, or bug in another package that `stantargets` depends on.
* [x] If there is [already a relevant issue](https://github.com/ropensci/stantargets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) like [this one](https://github.com/ropensci/targets/issues/256#issuecomment-754229683) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [x] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [x] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

`tar_stan_compile` does not include a repository argument. When the default repository is either `"aws"` or `"gcp"` calls fail with "_store_ the condition has length > 1"

## Expected result

Model compiles successfully & output is stored.

I believe the intention is for the `tar_target_raw` within `tar_stan_compile` to use `repository = "local"` (like internal calls for `name_file` in `tar_stan_mcmc`). Adding this argument to the relevant`tar_target_raw` fixes the problem, yet I can't quite work out why it shouldn't work with `tar_target_raw` autodetecting the repository, so I've not submitted the PR which would fix it for fear that I haven't solved the root cause.

## Diagnostic information

* Stantargets: 092b0b9
* Targets: ropensci/targets@f37af16

## Reproducible example

``` r
library(targets)
tar_script({
    stanfile <- "model.stan"
    tar_option_set(
        resources = tar_resources(
            gcp = tar_resources_gcp(bucket = "my-gcp-bucket"), 
            aws = tar_resources_aws(bucket = "my-aws-bucket")
        ), 
        cue = tar_cue(mode = "always"), 
        format = "qs", 
        repository = "gcp"  # Same error with "aws"
    )
    cmdstanr::write_stan_file("\n        data {\n          int<lower=1> N;\n          vector[N] x;\n          vector[N] y;\n        }\n\n        parameters {\n          real alpha;\n          real beta;\n          real<lower=0> sigma;\n        }\n\n        model {\n          alpha ~ std_normal();\n          beta ~ std_normal();\n          sigma ~ std_normal();\n          y ~ normal(alpha + x * beta , sigma);\n        }\n    ", 
        dir = getwd(), 
        basename = stanfile)
    list(
        stantargets::tar_stan_compile(
            model, 
            stanfile
        )
    )
})
tar_make()
#> • start target model
#> ✖ error target model
#> • end pipeline: 8.632 seconds
#> Error : _store_ the condition has length > 1
#> ✖ Problem with the pipeline.
#> Error:
#> ! problem with the pipeline.
```

<sup>Created on 2022-08-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

```
R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /apps/R/4.2.0/lib64/R/lib/libRblas.so
LAPACK: /apps/R/4.2.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8      
 [8] LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] stantargets_0.0.5.9000         fstcore_0.9.12                 testthat_3.1.4                 future.batchtools_0.10.0-9000  batchtools_0.9.16              future_1.26.1                 
 [7] tarchetypes_0.6.0.9000         rlang_1.0.4                    googleCloudStorageR_0.7.0.9000 dplyr_1.0.9                    targets_0.12.1.9000           

loaded via a namespace (and not attached):
  [1] backports_1.4.1        bigrquery_1.4.0        googleAuthR_2.0.0.9000 igraph_1.3.2           paws.common_0.3.17     lazyeval_0.2.2         RApiSerialize_0.1.0    listenv_0.8.0         
  [9] usethis_2.1.6          ggplot2_3.3.6          inline_0.3.19          digest_0.6.29          htmltools_0.5.2        fansi_1.0.3            magrittr_2.0.3         checkmate_2.1.0       
 [17] memoise_2.0.1          covr_3.5.1             base64url_1.4          tzdb_0.3.0             remotes_2.4.2          globals_0.15.1         readr_2.1.2            RcppParallel_5.1.5    
 [25] matrixStats_0.62.0     R.utils_2.12.0         askpass_1.1            prettyunits_1.1.1      colorspace_2.0-3       rappdirs_0.3.3         xfun_0.31              callr_3.7.1           
 [33] crayon_1.5.1           jsonlite_1.8.0         roxygen2_7.2.0         stringfish_0.15.7      brew_1.0-7             glue_1.6.2             xmlparsedata_1.0.5     gtable_0.3.0          
 [41] gargle_1.2.0           V8_4.2.0               distributional_0.3.0   R.cache_0.15.0         pkgbuild_1.3.1         rstan_2.26.13          abind_1.4-5            scales_1.2.0          
 [49] DBI_1.1.3              Rcpp_1.0.9             progress_1.2.2         bit_4.0.4              praise_1.0.0           clisymbols_1.2.0       stats4_4.2.0           StanHeaders_2.26.13   
 [57] xopen_1.0.0            rex_1.2.1              httr_1.4.3             posterior_1.2.2        ellipsis_0.3.2         pkgconfig_2.0.3        loo_2.5.1              R.methodsS3_1.8.2     
 [65] farver_2.1.1           dbplyr_2.2.1           utf8_1.2.2             tidyselect_1.1.2       labeling_0.4.2         munsell_0.5.0          tools_4.2.0            cachem_1.0.6          
 [73] cli_3.3.0              generics_0.1.3         devtools_2.4.3         evaluate_0.15          stringr_1.4.0          fastmap_1.1.0          paws_0.1.12            yaml_2.3.5            
 [81] processx_3.7.0         knitr_1.39             bit64_4.0.5            fs_1.5.2               zip_2.2.0              purrr_0.3.4            mime_0.12              R.oo_1.25.0           
 [89] xml2_1.3.3             brio_1.1.3             compiler_4.2.0         rstudioapi_0.13        curl_4.3.2             reprex_2.0.1           tibble_3.1.7           stringi_1.7.8         
 [97] highr_0.9              cyclocomp_1.1.0        ps_1.7.1               desc_1.4.1.9000        whoami_1.3.0           styler_1.7.0           paws.storage_0.1.12    commonmark_1.8.0      
[105] goodpractice_1.0.3     tensorA_0.36.2         vctrs_0.4.1            pillar_1.7.0           lifecycle_1.0.1        data.table_1.14.2      rcmdcheck_1.4.0        R6_2.5.1              
[113] qs_0.25.3              gridExtra_2.3          parallelly_1.32.0      sessioninfo_1.2.2      codetools_0.2-18       assertthat_0.2.1       pkgload_1.3.0          openssl_2.0.2         
[121] rprojroot_2.0.3        withr_2.5.0            parallel_4.2.0         hms_1.1.1              lintr_3.0.0            fst_0.9.8              grid_4.2.0             tidyr_1.2.0           
[129] waldo_0.4.0            rmarkdown_2.14         cmdstanr_0.5.2.1       lubridate_1.8.0    
```

**Comment by wlandau** [comment]
The intent of `tar_stan_compile()` is to compile the model on the local file system where the model is meant to run, so I consider cloud storage to be out of scope. Other target factories have a `compile` argument that allows you to choose whether to compile the original local model file or to compile a copy on wherever the parallel worker lives. The error you reproduced happens because the `tar_stan_compile()` target returns a vector of two file paths (the source and the compiled object) and `format = "file"` for cloud storage requires that the returned file path just be a single file and not a directory or collection of files. (Just emphasized this more in the `tar_target()` Rd file.)

**Comment by stuvet** [comment]
Just pointing out that I can't use `tar_stan_compile` when I set a default _pipeline-level_ `repository="gcp"/"aws"` - it will always give me this error.

Since there is no `repository` argument in `tar_stan_compile`, the only way around it is to set the default _pipeline-level_ repository to "local". Should the internal call to `tar_targets_raw` have a `repository="local"` instead?

**Comment by wlandau** [comment]
Ah, got it. Fixed in 4840334a7d964ea0ba098009f702e0aca0c5a329.
